### PR TITLE
chore: Add back tests

### PR DESCRIPTION
**Pull request by AndyUB**
## What is done?

The tests that were broken are added back.

## What is missing?

1. How to detect hang in cuda_stream.sync()?
2. Prevent a bound node from allreduce to be bound again.
3. Check inputs are tensors.
4. Update comments.
5. Update PR docs.
6. Any missing (important) test cases?

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/dag/tests/experimental/test_torch_tensor_dag.py [hunk 1]
```diff
@@ -1799,44 +1799,44 @@ def test_torch_tensor_nccl_all_reduce_scheduling(ray_start_regular):
...
-# def test_torch_tensor_nccl_all_reduce_scheduling_one_ready(ray_start_regular):
-#     """
-#     Test that scheduling avoids deadlocks when allreduce is used.
-#     """
-#     if not USE_GPU:
-#         pytest.skip("NCCL tests require GPUs")
+@pytest.mark.parametrize("ray_start_regular", [{"num_cpus": 4}], indirect=True)
+def test_torch_tensor_nccl_all_reduce_scheduling_one_ready(ray_start_regular):
+    """
+    Test that scheduling avoids deadlocks when allreduce is used.
```
Comment (dengwxn) [inline python/ray/dag/tests/experimental/test_torch_tensor_dag.py:1805]: Update comments to explain the potential deadlock.

File: python/ray/dag/tests/experimental/test_execution_schedule.py [hunk 2]
```diff
@@ -288,100 +288,116 @@ def test_two_ready_nccl_collectives(self, monkeypatch):
...
-    #     )
-    #     mock_graph[dag_idx_3][_DAGNodeOperationType.COMPUTE].set_collective_edges(
-    #         {(dag_idx_2, _DAGNodeOperationType.COMPUTE)}
-    #     )
-    #     next_nodes = _select_next_nodes(mock_actor_to_candidates, mock_graph)
-    #     assert len(next_nodes) == 1
-    #     assert next_nodes[0] == mock_graph[dag_idx_1][_DAGNodeOperationType.COMPUTE]
+    def test_nccl_collectives_one_ready_one_partial(self, monkeypatch):
+        """
+        Simulate the case where the only candidates are NCCL collective
```
Comment (dengwxn) [inline python/ray/dag/tests/experimental/test_execution_schedule.py:316]: Update comments to match the test case.

File: python/ray/dag/tests/experimental/test_execution_schedule.py [hunk 3]
```diff
@@ -288,100 +288,116 @@ def test_two_ready_nccl_collectives(self, monkeypatch):
...
+        operations. In this case, `_select_next_nodes` should return all
+        NCCL collective operations of the earliest-bound collective group
+        that is ready.
+
+        driver -> fake_actor_1.allreduce_1 -> driver
```
Comment (dengwxn) [inline python/ray/dag/tests/experimental/test_execution_schedule.py:321]: Do you need to update the graph description here?

File: python/ray/dag/tests/experimental/test_execution_schedule.py [hunk 4]
```diff
@@ -288,100 +288,116 @@ def test_two_ready_nccl_collectives(self, monkeypatch):
...
-    #     mock_graph[dag_idx_2][_DAGNodeOperationType.COMPUTE].set_collective_edges(
-    #         {(dag_idx_3, _DAGNodeOperationType.COMPUTE)}
-    #     )
-    #     mock_graph[dag_idx_3][_DAGNodeOperationType.COMPUTE].set_collective_edges(
-    #         {(dag_idx_2, _DAGNodeOperationType.COMPUTE)}
-    #     )
-    #     next_nodes = _select_next_nodes(mock_actor_to_candidates, mock_graph)
-    #     assert len(next_nodes) == 1
-    #     assert next_nodes[0] == mock_graph[dag_idx_1][_DAGNodeOperationType.COMPUTE]
+    def test_nccl_collectives_one_ready_one_partial(self, monkeypatch):
```
Comment (dengwxn) [inline python/ray/dag/tests/experimental/test_execution_schedule.py:314]: Change to "one_ready_group".

File: python/ray/dag/tests/experimental/test_torch_tensor_dag.py [hunk 5]
```diff
@@ -1799,44 +1799,44 @@ def test_torch_tensor_nccl_all_reduce_scheduling(ray_start_regular):
...
 
-# @pytest.mark.parametrize("ray_start_regular", [{"num_cpus": 4}], indirect=True)
-# def test_torch_tensor_nccl_all_reduce_scheduling_one_ready(ray_start_regular):
-#     """
-#     Test that scheduling avoids deadlocks when allreduce is used.
-#     """
-#     if not USE_GPU:
-#         pytest.skip("NCCL tests require GPUs")
+@pytest.mark.parametrize("ray_start_regular", [{"num_cpus": 4}], indirect=True)
+def test_torch_tensor_nccl_all_reduce_scheduling_one_ready(ray_start_regular):
```
Comment (dengwxn) [inline python/ray/dag/tests/experimental/test_torch_tensor_dag.py:1803]: Change to "one_ready_group".

**Review by dengwxn — COMMENTED** [review]
Can you also check all the tests to make sure all the comments and graphs match the test case?

File: python/ray/dag/tests/experimental/test_execution_schedule.py (hunk 4 above)
Comment (AndyUB) [reply python/ray/dag/tests/experimental/test_execution_schedule.py:314]: ack

File: python/ray/dag/tests/experimental/test_execution_schedule.py (hunk 3 above)
Comment (AndyUB) [reply python/ray/dag/tests/experimental/test_execution_schedule.py:321]: done

File: python/ray/dag/tests/experimental/test_execution_schedule.py (hunk 2 above)
Comment (AndyUB) [reply python/ray/dag/tests/experimental/test_execution_schedule.py:316]: done

File: python/ray/dag/tests/experimental/test_torch_tensor_dag.py (hunk 1 above)
Comment (AndyUB) [reply python/ray/dag/tests/experimental/test_torch_tensor_dag.py:1805]: done

File: python/ray/dag/tests/experimental/test_torch_tensor_dag.py (hunk 5 above)
Comment (AndyUB) [reply python/ray/dag/tests/experimental/test_torch_tensor_dag.py:1803]: ack

**Pull request closed without merging**